crates/server/src/server/h2_handler.rs

```diff
@@ -25,6 +26,7 @@ use crate::{
 };
 
 pub(crate) async fn h2_handler<T, I>(
+    access: Arc<Access>,
```

Comment:
Instead of this (which requires making all of the protocol implementations aware of this), could there be a wrapper that implements `RequestHandler` and stores another `R: RequestHandler` insight? That seems like a better design option here.

Reply:
let me consider this, see if that cleans things up.

Reply:
Ok, before I implement this, I think there is another option that could change how to approach this. The current implementation handles the request packet, but will only after respond with an error. This allows us to incorporate the original Query and respond properly.

The other option would be to reject the query and drop the connection (no response). @andygabby, do you know what would be the preference here? dropping the request with no response would be "cheaper" than parsing the packet and responding properly. It might be more desirable if we're trying to handle a DOS from a specific address or network to not go through all the parsing and/or response logic.

Reply:
I thought I updated this comment. I've changed this to return earlier and not respond with an error message. This will ensure that we're completely dropping requests in the case that the ACL is denied. I think this is better for uses like trying to block some type of a DOS.